Validate uploaded file types against the expected types

sergeleger opened the pull request:
<!-- PR template: please provide enough information to guide your reviewers.
Please read Contributing.md before submitting a PR.  -->

### PR Goal? <!-- Explain the main objective of this PR. -->

Validates the user uploaded file type against the list of expected file types.  


### Fixes? <!-- List any issues this PR fixes, e.g. Fixes #42, Fixes #324 -->

#413 

### Feedback sought? <!-- What should reviewers focus on in particular? -->

General functionality impression, review of the new error messages. Opinions on the read along web component error messages (non-toaster) while in the studio web interface; I feel the UX is a bit jarring. 


### Priority? <!-- How soon would you like this PR reviewed, does it block other work? -->

Medium, non blocker.

### Tests added? <!-- Make sure your PR includes automated tests for your changes. -->

Yes. 

### How to test? <!-- Explain how reviewers should test this PR. -->

- Attempt to upload a non-image file to the read along component.
- Attempt to upload a non-text or non-xml file in Step 1 of Studio Web
- Attempt to upload a non-audio file in Step 1 of Studio Web
- Bad UX? Attempt to upload a non-image file in Step 2 of Studio Web
- Attempt to upload a non-HTML file in studio web's Editor component.

### Confidence? <!-- How confident are you that these changes are ready to merge? -->

High. I've refactored some of the code to properly integrate this functionality. 


### Version change? <!-- Do you think this PR should trigger a Major (Breaking Change)/Minor (New Feature)/patch (refactor/bug fix) version change? -->


<!-- Add any other relevant information here -->

joanise commented on packages/studio-web/src/i18n/messages.fr.json:
```diff
@@ -170,9 +171,11 @@
     "7997459583873215257": "Pas d'audio",
     "5640828320811588897": "Prière de préparer votre texte et votre audio et de choisir une langue.",
     "7065107025201081158": "Formulaire incomplet",
+    "6817170311264143962": "The file \"{$fileName}\" n'est pas un fichier audio compatible.",
```
"Le fichier"

joanise commented on packages/studio-web/src/i18n/messages.fr.json:
```diff
@@ -170,9 +171,11 @@
     "7719309746449095739": "Fichier ",
     "1326685349515945581": " lu, mais pas téléversé. Votre audio restera sur votre ordinateur.",
     "6899344040225872362": "Bravo!",
+    "968476464320510530": "The file \"{$fileName}\" n'est pas un fichier texte compatible.",
```
"Le fichier"

joanise commented on packages/studio-web/src/i18n/messages.es.json:
```diff
@@ -170,9 +171,11 @@
     "7997459583873215257": "No hay audio",
     "5640828320811588897": "Por favor seleccione o escriba el texto, seleccione o grabe el audio y seleccione el idioma.",
     "7065107025201081158": "Plantilla incompleta",
+    "6817170311264143962": "Fichero \"{$fileName}\" no es fichero de audio compatible.",
```
"Il fichero ... no es un fichero ..."

joanise replied:
Oops, not `Il`, but `El fichero`.

joanise commented on packages/studio-web/src/i18n/messages.es.json:
```diff
@@ -170,9 +171,11 @@
     "7719309746449095739": "Fichero ",
     "1326685349515945581": " procesado pero no cargado. Su audio se mantendrá en su computadora.",
     "6899344040225872362": "¡Genial!",
+    "968476464320510530": "Fichero \"{$fileName}\" no es fichero de texto compatible.",
```
"El fichero ... no es un fichero ..."

joanise commented on packages/web-component/src/components/read-along-component/error-messge.tsx:
```diff
@@ -0,0 +1,68 @@
+import { Component, Prop, h, Event, EventEmitter, State } from "@stencil/core";
```
Typo in the name of this file -- better call it `error-message.tsx`.

joanise commented on packages/web-component/src/i18n/messages.eng.json:
```diff
@@ -6,6 +6,7 @@
     "loading-error": "Error: the <FILETYPE> file \"<FILENAME>\" could not be loaded.",
     "parse-error": "Error: the <FILETYPE> file \"<FILENAME>\" could not be parsed.",
     "alignment-error": "Error: No alignments were found.",
+    "image-error": "Error: the \"<FILENAME>\" is not an image file.",
```
`... the file \"<FILENAME ...`

sergeleger committed:
style: style: vertically aligned alert messages

sergeleger committed:
fix: added file upload validation to the web component

sergeleger committed:
fix: validate all uploaded files in studio web

sergeleger committed:
fix: when a file upload is cancelled the data was still available to readalong

sergeleger committed:
test: removed input[type=file].click events from playwright test

They are not required, and Chrome Desktop/Mobile generates 3 file upload events
instead of just 2 as intended by the test. One of the events sets the text input
field to null and fails the test.

joanise commented on packages/studio-web/src/i18n/messages.es.json:
```diff
@@ -170,9 +171,11 @@
     "7997459583873215257": "No hay audio",
     "5640828320811588897": "Por favor seleccione o escriba el texto, seleccione o grabe el audio y seleccione el idioma.",
     "7065107025201081158": "Plantilla incompleta",
+    "6817170311264143962": "El fichero \"{$fileName}\" no es fichero de audio compatible.",
```
I guess my review comment was not clear enough, sorry: no es **un** fichero

joanise commented on packages/studio-web/src/i18n/messages.es.json:
```diff
@@ -14,6 +14,7 @@
     "3903003521039334232": " Seleccione un documento HTML offline de ReadAlong. ",
     "5192585697967841844": " Para obtener este documento, debería haber creado un ReadAlong utilizando el {$START_LINK}{$START_TAG_MAT_ICON}launch{$CLOSE_TAG_MAT_ICON} Studio{$CLOSE_LINK} y luego haber seleccionado «HTML offline» como formato. O puede utilizar el archivo HTML offline incluido en el formato de descarga Archivos web comprimidos. ",
     "2136422915588832930": "Barra de herramientas de audio",
+    "7149012067459406545": "Fichero \"{$fileName}\" no es fichero HTML.",
```
Somehow I missed this one on my earlier review: **El** fichero ... no es **un** fichero ...

joanise commented on packages/studio-web/src/i18n/messages.es.json:
```diff
@@ -170,9 +171,11 @@
     "7719309746449095739": "Fichero ",
     "1326685349515945581": " procesado pero no cargado. Su audio se mantendrá en su computadora.",
     "6899344040225872362": "¡Genial!",
+    "968476464320510530": "El fichero \"{$fileName}\" no es fichero de texto compatible.",
```
Ditto, no es **un** fichero

joanise reviewed (approved):
Small changes required to the Spanish messages, but otherwise this is ready to merge. No need to re-review this time, just make the fixes and merge.
